apps/studio/src/server/modules/page/__tests__/page.router.test.ts

```diff
@@ -0,0 +1,1206 @@
+import type { IsomerSchema } from "@opengovsg/isomer-components"
+import type { z } from "zod"
+import { TRPCError } from "@trpc/server"
+import { omit, pick } from "lodash"
+import { resetTables } from "tests/integration/helpers/db"
+import {
+  applyAuthedSession,
+  applySession,
+  createMockRequest,
+} from "tests/integration/helpers/iron-session"
+import {
+  setupFolder,
+  setupPageResource,
+  setupSite,
+} from "tests/integration/helpers/seed"
+
+import type { reorderBlobSchema, updatePageBlobSchema } from "~/schemas/page"
+import { createCallerFactory } from "~/server/trpc"
+import { db } from "../../database"
+import { pageRouter } from "../page.router"
+import { createDefaultPage } from "../page.service"
+
+const createCaller = createCallerFactory(pageRouter)
+
+describe("page.router", async () => {
+  let caller: ReturnType<typeof createCaller>
+  const session = await applyAuthedSession()
+
+  beforeAll(() => {
+    caller = createCaller(createMockRequest(session))
+  })
+
+  describe("readPage", () => {
+    beforeEach(async () => {
+      await resetTables("Site")
```

Comment:
actually this isn't quite right - we should also reset the `Resource` table to avoid inter-test pollution right? 

we don't want to have a later test being reliant on an earlier one